docs/notebooks/how_to_build_and_optimize_qaoa_using_swap_strategy.ipynb

```diff
@@ -0,0 +1,742 @@
BJCUlGR2JJEHphlK+Y9Tp05Rs2ZNChYsyMaNG8mdO7fZkTJUcnIyTz31FLVq1WLOnDlmxxGRLGLp0qW0bduWgIAAZs6ciYuL5nzEeehvq/xHkSJFWLZsGUeOHCEgICDLbSe0bNkyjh07Rt++fc2OIiJZSMuWLfnhhx+YPXs2QUFBaL5HnIlmKOWuVq1aRfPmzenVq1eWOnKwfv36JCYmsmXLFrOjiEgWFBwcTPfu3Xn33Xf59NNPtcuEOAU3swOI4/L19WXy5Mm88cYblC5dmj59+pgdye527drFhg0bWLhwodlRRCSL6tatG/Hx8fTr1w9PT0/ef/99syOJ3JcKpdxTz549OXLkCP369aNEiRK0atXK7Eh2NX78eIoWLUqbNm3MjiIiWVjfvn2Ji4vj/fffx9PTU6d1icNToZT7+vTTTzl27BgBAQFs3LiRatWqmR3JLs6ePcvcuXMZPnw4bm76pyEi5ho6dChxcXH0798fT09PunXrZnYkkbvSV025LxcXF77//nvq169Py5YtiY6OpmjRombHMtw333yDm5sb3bt3NzuKiAgWi4UxY8YQHx9Pjx49yJUrFx07djQ7lsgdaVGOpNvff/+Nj48PefLkYfPmzeTJk8fsSIZJSEigePHitG3bVsdPiohDSUlJoWvXrsydO5dFixbRsmVLsyOJ/Ie2DZJ0e+SRR1ixYgUnTpygffv2mWrz3QULFvD3339niYVHIuJcXFxcCA4OplWrVrz00kusXbvW7Egi/6EZSnlga9as4cUXX6Rbt25MmTLF6be0sFqtVKtWjUceeYT//e9/ZscREbmjhIQEWrduzcaNG4mIiKB27dpmRxK5STOU8sAaNWrEN998w9SpUxk3bpzZcWy2efNmfvnlF62iFBGHlj17dkJDQ6levTrNmjVj586dZkcSuUkzlPLQhgwZwpgxYwgJCaFt27Zmx3lo7dq14+DBg+zfv9/pZ1tFJPO7fPkyjRo14tixY/z000+UL1/e7EgiKpTy8FJSUggMDGTJkiVs2LCBGjVqmB3pgR0/fpxSpUrd3MBdRMQZxMbGUq9ePc6fP8+mTZt48sknzY4kWZwKpdjk+vXrNGzYkCNHjhAdHU2JEiXMjvRA3nnnHWbMmMHJkyfJlSuX2XFERNLtzJkz1K1bl6SkJDZt2sTjjz9udiTJwvQMpdgkR44chIWFkTt3bpo3b87FixfNjpRucXFxTJ8+nddff11lUkSczqOPPsqaNWtITk6mUaNGnD171uxIkoWpUIrNChUqxIoVKzh9+jT+/v7cuHHD7EjpMmvWLOLj4wkKCjI7iojIQylWrBhr1qzhwoUL+Pr6OtU39ZK5qFCKIcqWLcuiRYvYuHEjb775Jo7+JEVKSgoTJkygXbt2mfLUHxHJOp566inWrFnDiRMnePHFF4mPjzc7kmRBKpRimHr16jF9+nSCg4P59NNPzY5zT//73//47bfftFWQiGQK3t7erFy5kv379+Pn58f169fNjiRZjBbliOE++ugjPv74YxYsWED79u3NjnNHjRs35vLly0RFRWmrIBHJNDZu3EjTpk1p2LAhixYtwt3d3exIkkWoUIrhrFYrnTt35scff2T9+vXUqlXL7Ej/sm/fPipWrMjcuXMJCAgwO46IiKFWrVpFy5Ytadu2LXPmzMHV1dXsSJIFqFCKXSQkJNC4cWMOHjxIVFQUpUqVMjvSTT169GDFihUcP35c372LSKa0aNEi2rdvT5cuXZg2bRouLnrCTexLf8PELrJnz87ixYvJnz8/zZs35/z582ZHAiAmJobZs2cTFBSkMikimVbbtm357rvvmDFjBv3793f4hZLi/FQoxW68vLxYsWIFMTExtG3blsTERLMj8e233wLw+uuvm5xERMS+OnXqxNdff82ECRP44IMPzI4jmZwKpdhV6dKlCQsLIzIykh49epj6XXJiYiKTJ0+mU6dOFCxY0LQcIiIZ5c033+Szzz5jxIgRfPbZZ2bHkUzMzewAkvnVqVOHmTNnEhgYSOnSpXn//fdNyRESEsKpU6fo06ePKdcXETHDu+++S1xcHO+99x65c+emV69eZkeSTEiFUjJEQEAAR48eZdiwYTz55JO8/PLLGXp9q9XKV199RaNGjfD29s7Qa4uImO3jjz8mLi6OoKAgcufOTefOnc2OJJmMCqVkmCFDhnDkyBFee+01ihUrxvPPP59h146KimLbtm0sW7Ysw64pIuIoLBYL48aNIz4+nq5du5IrVy7atWtndizJRLRtkGSoxMREmjZtyu7du4mMjKRMmTIPNY4VK+c4x1Wukkgi2ciGBx4UohAW/rtReYcOHfjll1/49ddftX2GiGRZycnJvPLKK4SGhhIeHs6LL75odiTJJFQoJcNduHCB2rVrk5SURGRkZLoWyMQQwzrWsYMdbGUrO9hBHHH/eT9PPKlGNWpQg2pUowENuHbyGiVLluSrr77irbfesscfSUTEady4cYN27doRERHBypUreeGFF8yOJJmACqWY4ujRo9SsWZMyZcqwZs0acuTI8Z/3sWIliigmM5kFLCCJJNxwI4mk+46f9n5uuFF2Z1mODzzO6bDTeOb2tMcfR0TEqVy/fp0WLVoQHR3N2rVrqVGjhtmRxMmpUIppoqKiqF+/Pm3atGHOnDn/OlM7nHCGMYx97Et3ibyrG4A7VKQiIxhBK1rZHl5ExMnFx8fj6+vLwYMH+emnn6hYsaLZkcSJ6WEyMU3NmjX54YcfmDdvHh9++CEAscQSSCCtac0BDgDYViYB/jkQZz/78cOPQAKJJda2MUVEnFzu3LlZvnw5JUqUoHHjxhw+fNjsSOLENEMppvvss89477336L22N3MbzOUiF0km2W7Xc8WVfORjOtNpTWu7XUdExBmcO3eOF154gfj4eDZt2kTx4sXNjiROSIVSTJdiTcEnzIftbbZjsVqwWuz/V9IFF1JIYRSjGMSgO64MFxHJKv766y+ef/55XF1d
```

Comment:
It's been agreed that, by convention, instead of calling the `transpile` function we should be calling the preset pass managers.

This actually give us a great excuse to showcase the `StagedPassmanager` by doing something like:
```python
from qiskit.transpiler import generate_preset_pass_manager

pass_manager = generate_preset_pass_manager(basis_gates=basis_gates, ...)
pass_manager.pre_init = PassManager([...])  # Definition of pass_manager_post
transpiled_qaoa_circ = pass_manager.run(qaoa_circ)
```

Reply:
Cool! I changed to use `generate_preset_pass_manager`.